spec_util/meld.go

```diff
@@ -292,7 +359,10 @@ func (m *melder) recordConflict(dst, src *pb.Data) error {
 
 	if arePrims && haveCompatibleTypes(dstPrim, srcPrim) {
 		// No conflict.  Merge primitive metadata.
-		m.meldPrimitive(dstPrim, srcPrim)
+		err := m.meldPrimitive(dstPrim, srcPrim)
+		if err != nil {
+			return err
+		}
```

Comment:
The callers of `recordConflict` all set `hasConflict = true` right before the call. But if we are able to merge primitves, then there is no actual conflict. Seems like we need to either push the merging of primitives into the callers, or return a signal from `recordConflict` that no conflict actually occurred.

Reply:
Even before this change, `recordConflict` would merge primitives in some cases without introducing a `oneof`.  Looks like the existing code uses `hasConflict` to avoid merging examples on conflicts, but `recordConflict` takes care of merging examples when there isn't a conflict.

I'm not going to refactor that as part of this PR.